.changeset/silver-peaches-pump.md

```diff
@@ -2,4 +2,4 @@
 "astro": patch
 ---
 
-Updated `renderEndpoint` implementation to handle `Response` with immutable headers.
+Updated `renderEndpoint` implementation to only append the reroute directive if the response HTTP status code is 404 or 500.
```

Comment:
```suggestion
Fixes an issue where using `Response.redirect` in an endpoint led to an error.
```
That perfectly describes what maintainers need to know about this change!

The changeset also gets merged into [the changelog](https://github.com/withastro/astro/blob/main/packages/astro/CHANGELOG.md), which end-users would read. They don't need to know that a function called `renderEndpoint` exists, for example.